source/features/clear-pr-merge-commit-message.tsx

```diff
@@ -3,19 +3,34 @@ import select from 'select-dom';
 import * as pageDetect from 'github-url-detection';
 
 import features from '../feature-manager';
+import {getRepo} from '../github-helpers';
+import {getBranches} from './update-pr-from-base-branch';
+import getDefaultBranch from '../github-helpers/get-default-branch';
 import onPrMergePanelOpen from '../github-events/on-pr-merge-panel-open';
 
-function init(): void | false {
+async function init(): Promise<void | false> {
 	const messageField = select('textarea#merge_message_field')!;
 	const originalMessage = messageField.value;
-	const deduplicatedAuthors = new Set();
+	const preservedContent = new Set();
 
 	// This method ensures that "Co-authored-by" capitalization doesn't affect deduplication
 	for (const [, author] of originalMessage.matchAll(/co-authored-by: ([^\n]+)/gi)) {
-		deduplicatedAuthors.add('Co-authored-by: ' + author);
+		preservedContent.add('Co-authored-by: ' + author);
 	}
 
-	const cleanedMessage = [...deduplicatedAuthors].join('\n');
+	// Preserve closing issues numbers when a PR is merged into a non-default branch since GitHub doesn't close them #4531
+	if (getBranches().base !== await getDefaultBranch()) {
+		for (const keyword of select.all('.comment-body .issue-keyword[aria-label^="This pull request closes"]')) {
+			const closingKeyword = keyword.textContent!.trim(); // Keep the keyword as-is (closes, fixes, etc.)
+
+			const issueNumberElement = keyword.nextElementSibling as HTMLAnchorElement;
+			const isCrossRepo = getRepo(issueNumberElement)!.nameWithOwner !== getRepo()!.nameWithOwner;
+			const issueNumber = isCrossRepo ? issueNumberElement.href : issueNumberElement.textContent!;
```

Comment:
What if we just get the whole URL?  `fixes https://GitHub.com/user/repo/issues/123`


It works regardless of the repo. I'm not particularly worried about this scenario to be honest because the users who want this specific behavior can just disable the feature and user GitHub's own repo config.

Fixing it is ok, but having a clean `fixes #124` isn't worth the complexity here.